### ISSUE TITLE
Unable to reset the editor

### ISSUE THREAD

**Issue by Zabi-Progton** [niuware/mui-rte#42]
I am trying to reset the editor using the following commands:

```
const editorState = EditorState.createEmpty();
const updateEditorState = EditorState.set(editorState);
```

I read the document for reset in the **_examples_** folder and it does not work.

Also in the demo I tried using the clear icon to remove text with and without selecting the text, it does not work,  can someone please look into this or help out, I really need this.

**Comment by niuware** [comment]
@Zabi-Progton Please check this component [API](https://github.com/niuware/mui-rte/blob/master/README.md) to see how to you can achieve your goal as is not possible to use `draft-js` API directly. I have checked and the [example](https://github.com/niuware/mui-rte/tree/master/examples/reset-value) works as expected.

**Comment by Zabi-Progton** [comment]
Can you please at least give a solution as to why, I am unable to reset the value of the field to empty, after i set the value using hooks, The example stated, shows how to reset a value to some other other can we have an example where the value is reset to empty. I really need some assistance on this, I have been stuck here for very long.

**Comment by niuware** [comment]
Just set `undefined` or an empty string to the value to reset it. Also take in mind the value that the component expects is a stringified `RawDraftContentState` object, do not pass an `EditorState` object.
```js
import React, { useState } from 'react'
import MUIRichTextEditor from '../..'

const ResetValue = () => {
    const [content, setContent] = useState(defaultContentStateAsString)

    return (
        <>
            <button onClick={() => setContent(undefined)}>Reset content</button>
            <MUIRichTextEditor 
                value={content} 
                onSave={save}
            />
        </>
    )
}

export default ResetValue
```

Also if you are using versions previous to 1.5.1 of this component, please check how to reset the content using a key: https://github.com/niuware/mui-rte/issues/17#issuecomment-531064920

Hopes this helps you out.

**Comment by Zabi-Progton** [comment]
Ok I did try setting undefined in the state which I provided as a value to the MUIRichTextEditor, let me explain my question, Take this scenerio, 
1. I have a MUIRichTextEditor, 
2. When I write something in it and click the add button, that whole content that I wrote in the editor gets converted to HTML string and gets added to an array.
3. Once added to the array, I want to clear out the MUIRichTextEditor, to an empty field. No matter what I am setting here, I still get the old value after adding the string to the array. It just doesn't clear out.

So, in simple terms, how do I programmatically clear out the field of a MUIRichTextEditor. I have tried with _createEmpty()_, I have tried with _null_ state, _undefined_ as your recommended, nothing works.

Please do tell If I am missing something here.

**Comment by Zabi-Progton** [comment]
I update to 1.8.0 and I am using this to empty the textfield, 

`const emptyState = ContentState.createFromText('')
  const resetValue = JSON.stringify(convertToRaw(emptyState))
  setContent(resetValue)`

And, this works,

Now the text seems to get cleared programmatically, but I have another problem here, on Updating to any version from 1.4.0, **_when i start writing in the editor, the cursor position remains same and everything I write is written backwards(reverse)_**, I am using a hooks, is this the reason I am getting into this problem?

**Comment by niuware** [comment]
@Zabi-Progton After resetting the value property, the editor continue working as expected, is there a way to replicate the situation you have? You can use either a class or a function component with hooks without any problem.

**Comment by Zabi-Progton** [comment]
I would like to add that I did not know that we can't make the RTE a controlled input as, it itself maintains its own state. (_please correct me if I am wrong here_). I also wanted to know if I can 
1. Disable the input.
2. Disable the multiline and instead use the **enter press** to do some function.

Can that be done? is it possible in this mui-rte editor?

**Comment by niuware** [comment]
> I would like to add that I did not know that we can't make the RTE a controlled input as, it itself maintains its own state. (_please correct me if I am wrong here_). 

You are correct.

> 1. Disable the input.
> 2. Disable the multiline and instead use the **enter press** to do some function.

Please check the [README](https://github.com/niuware/mui-rte/blob/master/README.md), both points you mention are possible with the current API of the `MUIRichTextEditor` component.

**Comment by Zabi-Progton** [comment]
I am able to disappear the MUIRichTextEditor, with the read-only field, i don't think there is a way to disable it in the MUIRichTextEditor API.

About point 2, I can't find anything as to how to use the keydown press(on enter) to do some function, can you help out or should I rather post such questions in stack overflow.

**Comment by niuware** [comment]
When setting the `readOnly` property `true`, the editor's input will be disabled and the toolbar will be hidden by default. Currently it is not possible to disable the editor and keep the toolbar visible. 

Please check the [events](https://github.com/niuware/mui-rte/blob/master/examples/events/index.tsx) example to know how to set the `onChange` event. This event is triggered every time the editor state is changed (for example: pressing a key).

**Comment by miguelangeltorresfp** [comment]
Would there be any way to reset the state to the initial value ?
The reset example works perfectly as long as you provide a different default value for the content to load, but it doesn't work if you try to use it to undo all changes.

**Comment by niuware** [comment]
> Would there be any way to reset the state to the initial value ?
> The reset example works perfectly as long as you provide a different default value for the content to load, but it doesn't work if you try to use it to undo all changes.

@miguelangeltorresfp You can use [this](https://github.com/niuware/mui-rte/issues/17#issuecomment-531064920) solution.

**Comment by chanceeakin** [comment]
> I updated to 1.8.0 and I am using this to empty the textfield,
> 
> ```
> const emptyState = ContentState.createFromText('')
>   const resetValue = JSON.stringify(convertToRaw(emptyState))
>   setContent(resetValue)
> ```
> 
> And, this works,
> 
> Now the text seems to get cleared programmatically, but I have another problem here, on Updating to any version from 1.4.0, **_when i start writing in the editor, the cursor position remains same and everything I write is written backwards(reverse)_**, I am using a hooks, is this the reason I am getting into this problem?

@Zabi-Progton: With the `MUIRichTextEditor` exposing only access to the content state via a stringified rawContentState, the selection state is out of sync with your reset contentState, hence the backward typing.

@niuware: A suggestion or two: it'd be beneficial to accept an `EditorState` as the value in lieu of a stringified contentState for the `value` prop, as this would provide a more flexible API for use cases where the component does need full control. Alternately, you could call `EditorState.moveSelectionToEnd()` in those instances where the currentContent is blank.
